src/pybind/mgr/cephadm/tests/test_migration.py

```diff
@@ -38,12 +39,84 @@ def test_service_ls(cephadm_module: CephadmOrchestrator):
             cephadm_module.cache.last_daemon_update['host2'] = datetime.now()
 
             cephadm_module.migration.migrate()
-            assert cephadm_module.migration_current == 1
+            assert cephadm_module.migration_current == 2
 
             out = [o.spec.placement for o in wait(cephadm_module, cephadm_module.describe_service())]
             assert out == [PlacementSpec(count=2, hosts=[HostPlacementSpec(hostname='host1', network='', name=''), HostPlacementSpec(hostname='host2', network='', name='')])]
 
 
+@mock.patch("cephadm.module.CephadmOrchestrator._run_cephadm", _run_cephadm('[]'))
+def test_migrate_service_id_one(cephadm_module: CephadmOrchestrator):
+    with with_host(cephadm_module, 'host1'):
+        cephadm_module.set_store(SPEC_STORE_PREFIX + 'mon.wrong',
+            json.dumps({
+                'spec': {
+                    'service_type': 'mon',
+                    'service_id': 'wrong',
+                    'placement': {
+                        'hosts': ['host1']
+                    }
+                },
+                'created': datetime.utcnow().strftime(DATEFMT),
+            }, sort_keys=True),
+        )
```

Comment:
add another test similar to this but for a service that requires a service_id (e.g. mds or nfs) ...